src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDiagnosticFactory.cs

```diff
@@ -199,5 +199,17 @@ public static RazorDiagnostic Create_ChildContentSetByAttributeAndBody(SourceSpa
         {
             return RazorDiagnostic.Create(ChildContentSetByAttributeAndBody, source ?? SourceSpan.Undefined, attribute);
         }
+
+        public static readonly RazorDiagnosticDescriptor ChildContentMixedWithExplicitChildContent =
+            new RazorDiagnosticDescriptor(
+                "BL9996",
+                () => "Unrecognized child content. When using explicit child content, all significant content must be inside a " +
+                "child content element such as '<ChildContent>'.",
```

Comment:
Not absolutely essential, but I wonder if we could make this easier to understand. Possible phrasing (if we have the necessary info):

```
Unexpected element 'div' inside component 'SuperTable'. The component 'SuperTable' accepts the following top-level items only: 'Header', 'Body', 'Footer'.
```

Is that possible, and does that match the intent of the diagnostic?